### PR TITLE
Fix cache definition

### PR DESCRIPTION
**Pull request by fabioperrella**
## Motivation

The way the "global cache" was being set, it was not the same one which was used inside the classes `Authentication::Cache` and `Client::Cache` because it used a class variable which was different for each class.

I could realize it debugging my Rails application. I was setting to use the Memory store but the cache files were being saved in filesystem (because it was using the `Rails.cache`).

## Proposed solution

Move the class variable `cache_backend` to `Cassette.cache_backend` and use it when necessary.

Doing this way I was forced to break the compatibility with the previous versions because the `cache_backend` will be set in a different way.

### CODE REVIEW

File: lib/cassette/cache.rb [hunk 1]
```diff
@@ -8,28 +8,6 @@
...
-          ::Rails.cache
-        elsif defined?(::ActiveSupport::Cache::MemoryStore)
-          ActiveSupport::Cache::MemoryStore.new
-        else
-          NullStore.new
-        end
-      end
-    end
-
-    def self.backend=(backend)
```
Comment (rhruiz) [inline lib/cassette/cache.rb]: Can't we delegate the writer just like we are delegating the reader?

Comment (fabioperrella) [reply lib/cassette/cache.rb]: You're right! I've just added this writer and now I could keep the compatibility. Could you review again pls?

**Comment by rhruiz** [comment]
@fabioperrella byebug broke que old ruby tests, maybe we should switch < 2.3 to use the Gemfile.old-ruby too

**Comment by fabioperrella** [comment]
> @fabioperrella byebug broke old ruby tests, maybe we should switch < 2.3 to use the Gemfile.old-ruby too

@rhruiz  now it is fixed!

I also updated the ruby version to the newest ones, ok?

**Pull request merged**